Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/FindPeaksAutomatic.py

```diff
@@ -0,0 +1,400 @@
+        self._estimate_peak_sigma = self.getProperty('EstimatePeakSigma').value
+        self._min_sigma = self.getProperty('MinPeakSigma').value
+        self._max_sigma = self.getProperty('MaxPeakSigma').value
+
+        if self._max_sigma < self._min_sigma:
+            issues['MinPeakSigma'] = 'Sigma bounds must be: MinPeakSigma <= MaxPeakSigma'
+            issues['MaxPeakSigma'] = 'Sigma bounds must be: MinPeakSigma <= MaxPeakSigma'
+
+        if self._estimate_peak_sigma < self._min_sigma:
+            issues['EstimatePeakSigma'] = 'EstimatePeakSigma must be greater than MinPeakSigma'
+
+        if self._estimate_peak_sigma > self._max_sigma:
+            issues['EstimatePeakSigma'] = 'EstimatePeakSigma must be greater than MaxPeakSigma'
+
+        return issues
+
+    def PyExec(self):
+        # Progress reporter for algorithm initialization
+        prog_reporter = Progress(self, start=0.0, end=0.1, nreports=4)
+
+        raw_xvals, raw_yvals, raw_error, error_ws = self.load_data(prog_reporter)
+
+        # Convert the data to point data
+        prog_reporter.report('Converting to point data')
+        raw_data_ws = ConvertToPointData(error_ws, StoreInADS=False)
+        raw_xvals = raw_data_ws.readX(0).copy()
+        raw_yvals = raw_data_ws.readY(0).copy()
+
+        raw_xvals, raw_yvals, raw_error = self.crop_data(raw_xvals, raw_yvals, raw_error, prog_reporter)
+
+        # Find the best peaks
+        (peakids, peak_table,
+         refit_peak_table), baseline = self.process(raw_xvals,
+                                                    raw_yvals,
+                                                    raw_error,
+                                                    acceptance=self._acceptance,
+                                                    average_window=self._smooth_window,
+                                                    bad_peak_to_consider=self._bad_peak_to_consider,
+                                                    use_poisson=self._use_poisson_cost,
+                                                    peak_width_estimate=self._estimate_peak_sigma,
+                                                    fit_to_baseline=self._fit_to_baseline,
+                                                    prog_reporter=prog_reporter)
+
+        if self._plot_peaks:
+            self.plot_peaks(raw_xvals, raw_yvals, baseline, peakids)
+
+        self.set_output_properties(peak_table, refit_peak_table)
+
+    def load_data(self, prog_reporter):
+        # Load the data and clean from Nans
+        raw_xvals = self.getProperty('InputWorkspace').value.readX(0).copy()
+        raw_yvals = self.getProperty('InputWorkspace').value.readY(0).copy()
+        prog_reporter.report('Loaded data')
+
+        # If the data does not have errors use poisson statistics create an workspace with added errors
+        raw_error = self.getProperty('InputWorkspace').value.readE(0).copy()
+        if len(np.argwhere(raw_error > 0)) == 0:
```

Comment:
One final tweak if we can. This currently runs CreateWorkspace as a parent algorithm and the workspace will go straight to the ADS. As you have done else where can we use the `StoreInADS=False` argument capture the variable returned and use the workspace handles and not names:

```python
            raw_error = np.sqrt(raw_yvals)
            error_ws = CreateWorkspace(DataX=raw_xvals,
                            DataY=raw_yvals,
                            DataE=raw_error,
                            StoreInADS=False)
        else:
            error_ws = self.getProperty('InputWorkspace').value

        self.setProperty('OutputWorkspace', error_ws)
        return raw_xvals, raw_yvals, raw_error, error_ws
```
This allows the user to specify the output name as in other algorithms but your default means they are not required to.